src/css3/new_tokenizer.rs

```diff
@@ -0,0 +1,163 @@
+}
+
+/// CSS Tokenizer according to the [w3 specification](https://www.w3.org/TR/css-syntax-3/#tokenization)
+pub struct Tokenizer<'stream> {
+    pub stream: &'stream mut InputStream,
+    /// Next token that has been consumed.
+    pub current_token: Option<String>,
+    /// Next token that has not yet been consumed
+    pub next_token: Option<String>,
+}
+
+impl<'stream> Tokenizer<'stream> {
+    pub fn new(stream: &'stream mut InputStream) -> Tokenizer {
+        Tokenizer {
+            stream,
+            current_token: None,
+            next_token: None,
+        }
+    }
+
+    /// 4.3.2. [Consume comments](https://www.w3.org/TR/css-syntax-3/#consume-comment)
+    pub fn consume_comment(&mut self) {
+        if self.stream.look_ahead_slice(2) == "/*" {
+            while self.stream.look_ahead_slice(2) != "*/" {
+                self.stream.read_char();
+            }
+
+            // consume '*/'
+            self.consume_chars(2);
+        };
+    }
+
+    /// 4.3.12. [Consume a number](https://www.w3.org/TR/css-syntax-3/#consume-number)
+    pub fn consume_number(&mut self) -> Number {
+        let mut value = String::new();
+        let mut kind = NumberKind::Integer;
+        let lookahead = self.stream.current_char().utf8();
+
+        if lookahead == '+' || lookahead == '-' {
+            value.push(self.stream.read_char().utf8());
+        }
+
+        value.push_str(&self.consume_digits());
+
+        if self.stream.current_char().utf8() == '.' && self.stream.next_char().utf8().is_numeric() {
+            value.push_str(&self.consume_chars(2));
+            kind = NumberKind::Number;
+        }
+
+        // type should be "number"
+        value.push_str(&self.consume_digits());
+
+        // todo: move them to gobal constants
+        // U+0045: LATIN CAPITAL LETTER E (E)
+        // U+0065: LATIN SMALL LETTER E (e)
+        if self.stream.current_char().utf8() == '\u{0045}'
+            || self.stream.current_char().utf8() == '\u{0065}'
+        {
+            value.push(self.stream.read_char().utf8());
+            kind = NumberKind::Number;
+
+            if self.stream.current_char().utf8() == '-' || self.stream.current_char().utf8() == '+'
+            {
+                value.push(self.stream.read_char().utf8());
+            }
+        }
+
+        value.push_str(&self.consume_digits());
+
+        Number::new(kind, value.parse().unwrap())
+    }
+
+    pub fn consume_digits(&mut self) -> String {
+        let mut value = String::new();
+
+        while self.stream.current_char().utf8().is_numeric() {
+            value.push(self.stream.read_char().utf8())
+        }
+
+        value
+    }
+
+    pub fn consume_chars(&mut self, mut len: usize) -> String {
+        let mut value = String::new();
+
+        while len > 0 {
+            println!("value_={:?}", self.stream.current_char().utf8());
+            value.push(self.stream.read_char().utf8());
+            len -= 1
+        }
+
+        println!("v={}", value);
```

Comment:
Did you intend to keep these `println!` statements?  Should we use `logger` instead?